cpp/src/server/handleMsg.cc

```diff
@@ -4,20 +4,50 @@ void handleMsg(
   const int currentClientFd,
   const ChatroomToFdList &chatRooms,
   const FdToName &clients,
-  string msg
+  string msg,
+  bool &personalMsg,
+  string &receiver
 ) {
   auto tokens = split(msg, DELIM, 2);
-  auto chatRoomName = tokens[0];
+  // If complete msg is not received in one recv call
+  if(tokens.size() != 2) {
+    if (personalMsg) {
+      auto receiverItr = find_if(
+        clients.begin(),
+        clients.end(),
+        [&receiver](auto const &itr) -> bool {
+          return itr.second == receiver;
+        });
+      send(receiverItr->first, msg.c_str(), msg.size(), 0);
+    }
+    else {
+      broadcast(chatRooms.at(receiver), currentClientFd, msg);
+    }
+    return;
+  }
+  // details have chatroom name and personal msg receiver's name
+  auto details = tokens[0];
+  string receiverName;
+  string chatRoomName;
+  //If msg is personal
+  auto index = details.find('@');
+  if(index != string::npos) {
```

Comment:
``` cpp
if(details.find('@') != string::npos) {
```

Reply:
We are using index variable again in if block.